src/test/java/org/cloudfoundry/promregator/discovery/CFMultiDiscovererTest.java

```diff
@@ -45,6 +55,18 @@ public void receiver(Instance instance) {
 	
 	@Test
 	public void testDiscoverWithCleanup() throws InterruptedException {
+		List<ResolvedTarget> resolvedTargets = new ArrayList<>();
+		ResolvedTarget aTarget = new ResolvedTarget();
+		aTarget.setOrgName("unittestorg");
+		aTarget.setSpaceName("unittestspace");
+		aTarget.setApplicationName("testapp");
+		aTarget.setApplicationId(CFAccessorMock.UNITTEST_APP1_UUID);
+		aTarget.setProtocol("https");
+		aTarget.setPath("/metrics");
+		aTarget.setOriginalTarget(new Target());
+		resolvedTargets.add(aTarget);
+		when(targetResolver.resolveTargets(any())).thenReturn(resolvedTargets);
```

Comment:
I replaced the `TrivialTargetResolver` with this mock and the tests are passing again